auacm/app/modules/competition_manager/views.py

```diff
@@ -197,57 +202,90 @@ def get_competition_data(cid):
     })
 
 
+@app.route('/api/competitions/<int:cid>', methods=['DELETE'])
+def delete_competition(cid):
+    """Delete a competition and all the data associated with it"""
+    competition = (database.session.query(Competition)
+                   .filter_by(cid=cid).first())
+
+    if not competition:
+        return '404: Competition not found', 404
+
+    # Delete the problems from a competition
+    competition_problems = (database.session.query(CompProblem)
+                            .filter_by(cid=cid).all())
+    for competition_problem in competition_problems:
+        database.session.delete(competition_problem)
+
+    # Delete the users/tams from a competition
+    competition_users = (database.session.query(CompUser)
+                         .filter_by(cid=cid).all())
+    for user in competition_users:
+        database.session.delete(user)
+
+    # Delete the competition itself
+    database.session.delete(competition)
+    database.session.commit()
+
+    return 'Successfully deleted', 204
```

Comment:
Why aren't we using `serve_response()` here?

Also, I have reservations about using the 204 response code. I know that *technically* it means "The server successfully processed the request and is not returning any content," however from my experience, it's more idiomatic to simply always use 200 for all responses that were successful. I'm open to leaving this as a 204, but it just seems odd.